pallets/gear/src/manager/mod.rs

```diff
@@ -177,12 +192,16 @@ where
             Default::default()
         };
 
+        let subsequent_execution = subsequent_execution || self.programs.contains(&id);
+        self.programs.insert(id);
```

Comment:
being in self.programs is not guarantee that we were executing this program this block: if we check some user id and its a program, we also append the id in programs

Reply:
one more thing - if we try to execute program and have no gas to load pages also means that further execution should try to do this too